src/gui/DatabaseTabWidget.cpp

```diff
@@ -756,17 +669,14 @@ void DatabaseTabWidget::lockDatabases()
         DatabaseWidget* dbWidget = static_cast<DatabaseWidget*>(widget(i));
         Database* db = databaseFromDatabaseWidget(dbWidget);
 
-        DatabaseWidget::Mode mode = dbWidget->currentMode();
-
-        if ((mode != DatabaseWidget::ViewMode && mode != DatabaseWidget::EditMode)
-                || !dbWidget->dbHasKey()) {
```

Comment:
@droidmonkey shouldn't we keep the `dbHasKey()` check to prevent locking?

Reply:
Added back in, good idea.